File: docs/snippets/all/archetypes/text_log.cpp
// Log a `TextLog`

#include <dalaran.hpp>

int main(int argc, char* argv[]) {
    const auto rec = dalaran::RecordingStream("dalaran_example_text_log");
    rec.spawn().exit_on_failure();

    rec.log(
        "log",
        dalaran::TextLog("Application started.")
            .with_level(dalaran::TextLogLevel::Info)
    );
}

File: docs/snippets/all/archetypes/text_log.rs
//! Log a `TextLog`

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec =
        dalaran::RecordingStreamBuilder::new("dalaran_example_text_log").spawn()?;

    rec.log(
        "log",
        &dalaran::TextLog::new("Application started.").with_level("INFO"),
    )?;

    Ok(())
}
